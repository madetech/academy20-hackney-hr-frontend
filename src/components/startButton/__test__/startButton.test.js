import React from 'react';
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import StartButton from "../startButton";
import renderer from 'react-test-renderer';

let container = null;
beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
});

afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null
});

it("renders with or without some text", () => {
    act(() => {
        render(<StartButton/>, container);
    });
    expect(container.textContent).toBe("Begin");
    act(() => {
        render(<StartButton text="Register your visit"/>, container);
    });
    expect(container.textContent).toBe("Register your visit");
    act(() => {
        render(<StartButton text="Employee login"/>, container);
    });
    expect(container.textContent).toBe("Employee login");
});

test('Renders button', () => {
    const component = renderer.create(
      <StartButton/>,
    );
    let test = component.toJSON();
    expect(test).toMatchSnapshot();
    
  });