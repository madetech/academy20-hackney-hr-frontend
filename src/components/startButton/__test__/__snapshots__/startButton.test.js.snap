// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders button 1`] = `
<button
  className="button-start"
>
  <p>
    Begin
  </p>
  <svg
    className="button-start_svg"
    height="19"
    viewBox="0 0 33 40"
    width="17.5"
  >
    <path
      d="M0 0h13l20 20-20 20H0l20-20z"
      fill="currentColor"
    />
  </svg>
</button>
`;
